independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/MultiResponseHandler.java

```diff
@@ -42,30 +43,65 @@ public Object apply(Object v, Throwable t) {
 
     private static class StreamingMultiSubscriber extends AbstractMultiSubscriber {
 
-        StreamingMultiSubscriber(ResteasyReactiveRequestContext requestContext) {
+        // Huge hack to stream valid json
+        private boolean json;
+        private String nextJsonPrefix;
+        private boolean hadItem;
+
+        StreamingMultiSubscriber(ResteasyReactiveRequestContext requestContext, boolean json) {
             super(requestContext);
+            this.json = json;
+            this.nextJsonPrefix = "[";
+            this.hadItem = false;
         }
 
         @Override
         public void onNext(Object item) {
-            StreamingUtil.send(requestContext, item).handle(new BiFunction<Object, Throwable, Object>() {
-                @Override
-                public Object apply(Object v, Throwable t) {
-                    if (t != null) {
-                        // need to cancel because the exception didn't come from the Multi
-                        try {
-                            subscription.cancel();
-                        } catch (Throwable t2) {
-                            t2.printStackTrace();
+            hadItem = true;
+            StreamingUtil.send(requestContext, item, json ? nextJsonPrefix : null)
+                    .handle(new BiFunction<Object, Throwable, Object>() {
+                        @Override
+                        public Object apply(Object v, Throwable t) {
+                            if (t != null) {
+                                // need to cancel because the exception didn't come from the Multi
+                                try {
+                                    subscription.cancel();
+                                } catch (Throwable t2) {
+                                    t2.printStackTrace();
+                                }
+                                handleException(requestContext, t);
+                            } else {
+                                // next item will need this prefix if json
+                                nextJsonPrefix = ",";
```

Comment:
Aren't you adding a `,` even for the last item?

Reply:
No, this is only used if we write an item, not at the end of the stream.

Reply:
oh sorry! Got it!